### PR TITLE
refactor 'fetch'

### PR DESCRIPTION
**Pull request by wmhilton**
This PR adds no features and changes nothing outwardly. It just refactors the insides some based on the `http-receiver` PR.

(I've given up on merging `http-receiver`. It's just too big a PR to maintain. Instead I am going to merge a little bit at a time.)

### CODE REVIEW

File: __tests__/test-GitRemoteConnection.js [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  GitRemoteConnection,
+  GitSideBand
+} = require('isomorphic-git/internal-apis')
+const bufferToStream = require('buffer-to-stream')
+const pify = require('pify')
+const concat = require('simple-concat')
+const stream = require('stream')
+
+/*
+A diagram might be helpful.
```
Comment (TomasHubelbauer) [inline __tests__/test-GitRemoteConnection.js]: `Server` is misaligned by one column to the left! 😢 *Important*

Comment (wmhilton) [reply __tests__/test-GitRemoteConnection.js]: Believe it or not, this is actually a mistake resulting from my editor font (Fira Code iScript) being slightly off of a true monospace. 😢

**Comment by wmhilton** [comment]
I think it keeps failing Travis because its a holiday in the US and their servers are busy.

**Comment by wmhilton** [comment]
Thought of a sly way to maybe make the tests pass by adding a feature!

**Pull request merged**